scrapy/core/downloader/handlers/ftp.py

```diff
@@ -70,10 +71,14 @@ class FTPDownloadHandler(object):
         "default": 503,
     }
 
-    def __init__(self, settings):
-        self.default_user = settings['FTP_USER']
-        self.default_password = settings['FTP_PASSWORD']
-        self.passive_mode = settings['FTP_PASSIVE_MODE']
+    def __init__(self, crawler):
```

Comment:
I think if you keep settings argument, instead of switching to crawler, then FTPDownloader subclasses which override `__init__` method would continue to work. The same applies to HTTP10DownloadHandler.